x-pack/plugins/index_management/public/application/sections/home/index_list/create_index/create_index_modal.tsx

```diff
@@ -0,0 +1,153 @@
+
+  const putCreateIndex = useCallback(async () => {
+    setIsSaving(true);
+    try {
+      const { error } = await createIndex(indexName);
+      setIsSaving(false);
+      if (!error) {
+        loadIndices();
+        closeModal();
+        return;
+      }
+      setCreateError(error.message);
+    } catch (e) {
+      setIsSaving(false);
+      setCreateError(e.message);
+    }
+  }, [closeModal, indexName, loadIndices]);
+
+  const onSave = () => {
+    if (isValidIndexName(indexName)) {
+      putCreateIndex();
+    }
+  };
+
+  const onNameChange = (name: string) => {
+    setIndexName(name);
+    if (!isValidIndexName(name)) {
+      setIndexNameError(INVALID_INDEX_NAME_ERROR);
+    } else if (indexNameError) {
+      setIndexNameError(undefined);
+    }
+  };
+
+  return (
+    <EuiModal onClose={closeModal} initialFocus="[name=indexName]">
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          <FormattedMessage
+            id="xpack.idxMgmt.createIndex.modal.title"
+            defaultMessage="Create index"
+          />
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        {createError && (
+          <>
+            <EuiCallOut
+              color="danger"
+              iconType="error"
+              title={i18n.translate('xpack.idxMgmt.createIndex.modal.error.title', {
+                defaultMessage: 'Error creating index',
+              })}
+            >
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.idxMgmt.createIndex.modal.error.description"
+                  defaultMessage="Error creating index: {errorMessage}"
+                  values={{ errorMessage: createError }}
+                />
+              </EuiText>
+            </EuiCallOut>
+            <EuiSpacer />
+          </>
+        )}
+        <EuiForm id="createIndexModalForm" component="form">
+          <EuiFormRow
+            fullWidth
+            label={i18n.translate('xpack.idxMgmt.createIndex.modal.indexName.label', {
+              defaultMessage: 'Index name',
+            })}
+            isDisabled={isSaving}
+            isInvalid={indexNameError !== undefined}
+            error={indexNameError}
+          >
+            <EuiFieldText
+              fullWidth
+              name="indexName"
+              value={indexName}
+              onChange={(e) => onNameChange(e.target.value)}
+            />
+          </EuiFormRow>
+        </EuiForm>
+      </EuiModalBody>
+      <EuiModalFooter>
+        <EuiButtonEmpty
+          onClick={closeModal}
+          disabled={isSaving}
+          data-telemetry-id="idxMgmt-indexList-createIndex-cancelButton"
+        >
+          <FormattedMessage
+            id="xpack.idxMgmt.createIndex.modal.cancelButton"
+            defaultMessage="Cancel"
+          />
+        </EuiButtonEmpty>
+        <EuiButton
+          fill
+          disabled={isSaving || indexNameError !== undefined}
```

Comment:
minor nit: I think we could use `isLoading` on the button while saving, instead of disabling it. With a longer running request, that makes it more clear that the request is loading